editor/src/components/editor/store/dispatch.tsx

```diff
@@ -369,13 +369,82 @@ export function resetDispatchGlobals(): void {
   accumulatedProjectChanges = emptyProjectChanges
 }
 
-export function editorDispatch(
+// With this reducer we can split the actions into groups (arrays) which can be dispatched together without rebuilding the derived state.
+// Between the different group derived state rebuild is needed
+function reducerToSplitToActionGroups(
+  actionGroups: EditorAction[][],
+  currentAction: EditorAction,
+  i: number,
```

Comment:
`i`?

Reply:
Oh I see this entire function was just moved

Reply:
Taken from the original code, I believe it's the index of the action.